internal/server/server.go

```diff
@@ -44,35 +46,40 @@ func Serve(
 	cfg Config,
 	nrApp newrelic.Application,
 	deps api.Deps,
-) (*server.MuxServer, error) {
-	s, err := server.NewMux(server.Config{
-		Port: cfg.Port,
-	}, server.WithMuxGRPCServerOptions(getGRPCMiddleware(cfg, logger, nrApp)))
-	if err != nil {
-		return nil, err
-	}
-
-	gw, err := server.NewGateway("", cfg.Port, server.WithGatewayMuxOptions(
-		runtime.WithIncomingHeaderMatcher(customHeaderMatcherFunc(map[string]bool{cfg.IdentityProxyHeader: true}))),
-	)
-	if err != nil {
-		return nil, err
-	}
+) error {
+	httpMux := http.NewServeMux()
 
-	registerHandler(ctx, s, gw, deps)
+	grpcGateway := runtime.NewServeMux(runtime.WithIncomingHeaderMatcher(customHeaderMatcherFunc(map[string]bool{cfg.IdentityProxyHeader: true})))
 
-	go s.Serve()
+	grpcServer := grpc.NewServer(getGRPCMiddleware(cfg, logger, nrApp))
+	reflection.Register(grpcServer)
 
-	logger.Info("[shield] api is up", "port", cfg.Port)
+	address := fmt.Sprintf("%s:%d", cfg.Host, cfg.GRPCPort)
+	err := registerHandler(ctx, httpMux, grpcGateway, grpcServer, deps, address)
+	if err != nil {
+		return err
+	}
 
-	return s, nil
-}
+	logger.Info("[shield] api server starting", "http-port", cfg.HTTPPort, "grpc-port", cfg.GRPCPort)
+
+	if err := mux.Serve(
+		ctx,
+		mux.WithHTTPTarget(fmt.Sprintf(":%d", cfg.HTTPPort), &http.Server{
+			Handler:        httpMux,
+			ReadTimeout:    120 * time.Second,
+			WriteTimeout:   120 * time.Second,
+			MaxHeaderBytes: 1 << 20,
+		}),
+		mux.WithGRPCTarget(fmt.Sprintf(":%d", cfg.GRPCPort), grpcServer),
+		mux.WithGracePeriod(5*time.Second),
+	); !errors.Is(err, context.Canceled) {
+		logger.Error("mux serve error", "err", err)
+		return err
```

Comment:
Any reason for both logging and returning the error? If the only intention for returning the error is logging, it is already taken care of here.

Reply:
Yup, my bad, returned with a `nil` error at this step.